api/defaults/defaults.go

```diff
@@ -145,4 +145,13 @@ const (
 const (
 	// TunnelPublicAddrEnvar optionally specifies the alternative reverse tunnel address.
 	TunnelPublicAddrEnvar = "TELEPORT_TUNNEL_PUBLIC_ADDR"
+
+	// TLSRoutingConnUpgradeEnvVar overwrites the test result for deciding if
+	// ALPN connection upgrade is required.
+	//
+	// Sample values:
+	// true
+	// <some.cluster.com>=yes,<another.cluster.com>=no
+	// 0,<some.cluster.com>=1
+	TLSRoutingConnUpgradeEnvVar = "TELEPORT_TLS_ROUTING_CONN_UPGRADE"
```

Comment:
I am thinking of not adding a cluster setting for this to keep it simple. The env var is just for last resource if something goes wrong, and debugging. What do you think? Even if we have a cluster setting, the overwrite is still useful.

Reply:
There is always a risk that the feature won' work because customer environment is quite complex/unpredicted setup but at the teleport onboarding state there are many tools that can be used instead for debugging for instance: 
```
openssl s_client -connect ice-berg.dev:3080  -servername ice-berg.dev -alpn teleport-reversetunnel
```

My main concerts is  that probably it will stay forever where this will be our undocumented internal mechanism to test connection upgrade. But wonder if we should keeping this temporary like TODO Remove in Teleport v14 or v15 ?